python/lsst/cp/pipe/deferredCharge.py

```diff
@@ -0,0 +1,1019 @@
+        """Generate a realization of the overscan model, using the specified
+        fit parameters and input signal.
+
+        Parameters
+        ----------
+        params : `lmfit.Parameters`
+            Object containing the model parameters.
+        signal : `np.ndarray`, (nMeasurements)
+            Array of image means.
+        num_transfers : `int`
+            Number of serial transfers that the charge undergoes.
+        amp : `lsst.afw.cameraGeom.Amplifier`
+            Amplifier to use for geometry information.
+        start : `int`, optional
+            First overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+        stop : `int`, optional
+            Last overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+        trap_type : `str`, optional
+            Type of trap model to use.
+
+        Returns
+        -------
+        results : `np.ndarray`, (nMeasurements, nCols)
+            Model results.
+        """
+        v = params.valuesdict()
+
+        # Adjust column numbering to match DM overscan bbox.
+        start += 1
+        stop += 1
+
+        # Electronics effect optimization
+        output_amplifier = FloatingOutputAmplifier(1.0, v['driftscale'], v['decaytime'])
+
+        # CTI optimization
+        v['cti'] = 10**v['ctiexp']
+
+        # Trap type for optimization
+        if trap_type is None:
+            trap = None
+        elif trap_type == 'linear':
+            trap = SerialTrap(v['trapsize'], v['emissiontime'], 1, 'linear',
+                              [v['scaling']])
+        elif trap_type == 'logistic':
+            trap = SerialTrap(v['trapsize'], v['emissiontime'], 1, 'logistic',
+                              [v['f0'], v['k']])
+        else:
+            raise ValueError('Trap type must be linear or logistic or None')
+
+        # Simulate ramp readout
+        imarr = np.zeros((signal.shape[0], amp.getRawDataBBox().getWidth()))
+        ramp = SegmentSimulator(imarr, amp.getRawSerialPrescanBBox().getWidth(), output_amplifier,
+                                cti=v['cti'], traps=trap)
+        ramp.ramp_exp(signal)
+        model_results = ramp.readout(serial_overscan_width=amp.getRawSerialOverscanBBox().getWidth(),
+                                     parallel_overscan_width=0)
+
+        ncols = amp.getRawSerialPrescanBBox().getWidth() + amp.getRawDataBBox().getWidth()
+
+        return model_results[:, ncols+start-1:ncols+stop]
+
+
+class SegmentSimulator:
+    """Controls the creation of simulated segment images.
+
+    Parameters
+    ----------
+    imarr : `np.ndarray` (nx, ny)
+        Image data array.
+    prescan_width : `int`
+        Number of serial prescan columns.
+    output_amplifier : `lsst.cp.pipe.BaseOutputAmplifier`
```

Comment:
I don't actually see `BaseOutputAmplifier` defined anywhere, and `FloatingOutputAmplifier` doesn't subclass anything and seems to be the object that quacks like that. Is that what you meant?